### PR TITLE
feat(bar chart): tooltip changes [CU-8695881ux]

### PR DESCRIPTION
**Pull request by joseazevedo88**
### Design
image

### Strorybook
image

#### Also fixed an issue where toggling the horizontal field in the bar chart stories wasn't working

### CODE REVIEW

**Review by joewoodson — CHANGES_REQUESTED** [review]
Code looks good, and the stories are working as expected, nice work! But I wonder if the name of the new prop could be made clearer -- what about `valueContext`, as the new content helps to contextualize the value and clarify its meaning?

**Review by joewoodson — APPROVED** [review]

**Comment by joseazevedo88** [comment]
@Frontify/frontend-platform I need a review/approval here, please. Thanks

File: packages/charts/src/components/BarChart/components/Axes.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const Axes = ({
     valueFormatter,
     labelFormatter,
 }: AxesProps) => {
-    const formatLabel = (label: string) => truncateTextLabel(labelFormatter(label));
+    const formatLabel = (label: string) => truncateTextLabel(labelFormatter(label).toString());
```
Comment (noahwaldner) [inline packages/charts/src/components/BarChart/components/Axes.tsx:41]: Why is this necessary? 

The typing of laybelFormatter is `export type LabelFormatter = (label: string) => string;` the value is already a string

File: packages/charts/src/components/common/components/Tooltip/Tooltip.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                },
+            },
+            tooltipOpen: true,
+        });
+        vi.mocked(usePositions).mockReturnValue({
+            tooltipPosition: { tooltipLeft: 20, tooltipTop: 30 },
+            glyphProps: [],
+        });
+        const { getByText } = render(
+            <Tooltip crossHairStyle="line" colorAccessor={(str) => str} valueContextBySeries={['%', 'ºC', 'F']} />,
```
Comment (noahwaldner) [inline packages/charts/src/components/common/components/Tooltip/Tooltip.spec.tsx:59]: if there is a case where no proper `colorAccessor` function is passed in here, it would make sense to make this optional and define a default in the Tooltip.

File: packages/charts/src/components/common/components/Tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -91,8 +93,11 @@ export const Tooltip = ({
         tooltipContext.tooltipData?.datumByKey,
         childSumLabel,
     );
+    const entriesWithFormattedSeries = valueContextBySeries
```
Comment (noahwaldner) [inline packages/charts/src/components/common/components/Tooltip/Tooltip.tsx:96]: You could move this into getTooltipEntries and potentially even memoize that function. That way you have one function that returns your formatted tooltip content

Comment (joseazevedo88) [reply packages/charts/src/components/common/components/Tooltip/Tooltip.tsx]: Moved, didn't do the memo though, no reason to do it unless it's slow as it also has a overhead cost.

File: packages/charts/src/components/common/components/Tooltip/Tooltip.spec.tsx (hunk 2 above)
Comment (joseazevedo88) [reply packages/charts/src/components/common/components/Tooltip/Tooltip.spec.tsx]: Made it optional with a fallback

File: packages/charts/src/components/BarChart/components/Axes.tsx (hunk 1 above)
Comment (joseazevedo88) [reply packages/charts/src/components/BarChart/components/Axes.tsx]: Fixed the typing, the `formatLabel` param is of type `string | number` in reality because of the axis toggle

File: packages/charts/src/components/common/components/Tooltip/Tooltip.tsx (hunk 3 above)
Comment (jose-costa-frontify) [reply packages/charts/src/components/common/components/Tooltip/Tooltip.tsx]: nothing is slow on a powerful enough cpu 😛 `.reverse` is efficient because it is performed in-place but i would suggest the memory cost of `useMemo` are more than surpassed by the memory spent spreading that array every time the component re-renders. Then it's off waiting for the garbage collector, potentially piles of lonely arrays with nothing pointing to them

**Review by jose-costa-frontify — APPROVED** [review]
LGTM
